### PR TITLE
mac_veriexec_parser: Fix open_file error handling

### PR DESCRIPTION
**Pull request by losfair**
NDFREE_PNBUF should be called after checking the return value of vn_open(), and should only be called once.

### CODE REVIEW

File: sys/security/mac_veriexec_parser/mac_veriexec_parser.c [hunk 1]
```diff
@@ -241,9 +241,9 @@ open_file(const char *path, struct nameidata *nid)
 
 	NDINIT(nid, LOOKUP, 0, UIO_SYSSPACE, path);
 	rc = vn_open(nid, &flags, 0, NULL);
-	NDFREE_PNBUF(nid);
 	if (rc != 0)
 		return (rc);
+	NDFREE_PNBUF(nid);
 
 	return (0);
```
Comment (ehem) [inline sys/security/mac_veriexec_parser/mac_veriexec_parser.c:248]: I would tend towards instead doing:
```
if (rc != 0)
        NDFREE_PNBUF(nid);

return (rc);
```
I don't know about other reviewers though.

**Review by ehem — APPROVED** [review]
Looks good to me.

**Review by gmshake — APPROVED** [review]
Looks good to me.

**Comment by gmshake** [comment]
CC @wojtas-marcin

**Comment by wojtas-marcin** [comment]
cc: @kornelduleba

File: sys/security/mac_veriexec_parser/mac_veriexec_parser.c (hunk 1 above)
Comment (kornelduleba) [reply sys/security/mac_veriexec_parser/mac_veriexec_parser.c:248]: Did you mean something like this?
```
if (rc == 0)
        NDFREE_PNBUF(nid);
```

The problem here is that if `vn_open` fails it'll free `cn_pnbuf` by itself, causing a double free in the error path.
We might want to do something like:
```
if (rc != 0)
	return (rc);
else
        NDFREE_PNBUF(nid);
```

but IMO the version that was submitted is fine as is.

**Comment by kornelduleba** [comment]
LGTM

**Comment by bsdimp** [comment]
lgtm. Will land.

**Pull request merged**